research/encoding_2d_data/decode.py

```diff
@@ -0,0 +1,23 @@
+import os
```

Comment:
Please add some text to decode.py explaining how data is being decoded, without having to read the entire code